File: BotsSDK/korebot/src/main/java/kore/botssdk/audiocodes/webrtcclient/Fragments/ChatFragment.java
package kore.botssdk.audiocodes.webrtcclient.Fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.audiocodes.mv.webrtcsdk.im.InstanceMessageStatus;

import kore.botssdk.R;
import kore.botssdk.application.BotApplication;
import kore.botssdk.audiocodes.webrtcclient.Callbacks.CallBackHandler;
import kore.botssdk.audiocodes.webrtcclient.General.ACManager;
import kore.botssdk.audiocodes.webrtcclient.General.AppUtils;

public class ChatFragment extends BaseFragment implements FragmentLifecycle {

    private static final String TAG = "ChatFragment";

    private TextView lastMessageContact;
    private TextView lastMessage;

    private CallBackHandler.ChatCallback chatCallback = new CallBackHandler.ChatCallback() {
        @Override
        public void onNewMessage(String user, String message) {
            handler.post(new Runnable() {
                @Override
                public void run() {
                    lastMessageContact.setText(user);
                    lastMessage.setText(message);
                }
            });
        }

        @Override
        public void onMessageStatus(InstanceMessageStatus instanceMessageStatus, long ID){
            handler.post(new Runnable() {
                @Override
                public void run() {
                    Toast.makeText(BotApplication.getGlobalContext(),
                            getString(R.string.chat_message_sent) + ": " + instanceMessageStatus,
                            Toast.LENGTH_SHORT).show();
                }
            });
        }
    };

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.main_fragment_chat, container, false);
        initGui(rootView);
        CallBackHandler.registerChatCallback(chatCallback);
        return rootView;
    }

    private void initGui(View rootView)
    {
        EditText contact = (EditText) rootView.findViewById(R.id.chat_contact_edittext);
        EditText message = (EditText) rootView.findViewById(R.id.chat_message_edittext);
        lastMessageContact = (TextView) rootView.findViewById(R.id.chat_last_message_contact_textview);
        lastMessage = (TextView) rootView.findViewById(R.id.chat_last_message_textview);
        Button sens = (Button) rootView.findViewById(R.id.chat_send_button);
        sens.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String user = contact.getText().toString();
                String text = message.getText().toString();
                if(!user.equals("") && !text.equals("")){
                    if (!ACManager.getInstance().isRegisterState()) {
                        Toast.makeText(getContext(), R.string.no_registration, Toast.LENGTH_SHORT).show();
                    }
                    else{
                        ACManager.getInstance().sendInstantMessage(user, text);
                    }
                }
            }
        });
        if(AppUtils.getStringBoolean(R.string.enable_debug_mode)){
            contact.setText(getString(R.string.sip_default_chat_contact));
            message.setText(getString(R.string.sip_default_chat_message));
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        CallBackHandler.unregisterChatCallback(chatCallback);
    }

    @Override
    public void onPauseFragment() {

    }

    @Override
    public void onResumeFragment() {

    }


}

File: BotSdk/ui/src/main/java/com/kore/ui/audiocodes/webrtcclient/fragments/ChatFragment.kt
package com.kore.ui.audiocodes.webrtcclient.fragments

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import com.audiocodes.mv.webrtcsdk.im.InstanceMessageStatus
import com.kore.ui.R
import com.kore.ui.audiocodes.webrtcclient.callbacks.CallBackHandler.ChatCallback
import com.kore.ui.audiocodes.webrtcclient.callbacks.CallBackHandler.registerChatCallback
import com.kore.ui.audiocodes.webrtcclient.callbacks.CallBackHandler.unregisterChatCallback
import com.kore.ui.audiocodes.webrtcclient.general.AppUtils
import com.kore.ui.databinding.MainFragmentChatBinding

@SuppressLint("UnknownNullness")
class ChatFragment : BaseFragment(), FragmentLifecycle {
    private var binding: MainFragmentChatBinding? = null

    private val chatCallback: ChatCallback = object : ChatCallback {
        override fun onNewMessage(user: String, message: String) {
            handler.post {
                binding?.apply {
                    chatLastMessageContactTextview.text = user
                    chatLastMessageTextview.text = message
                }
            }
        }

        override fun onMessageStatus(instanceMessageStatus: InstanceMessageStatus, ID: Long) {
            handler.post {
                Toast.makeText(requireContext(), getString(R.string.chat_message_sent) + ": " + instanceMessageStatus, Toast.LENGTH_SHORT)
                    .show()
            }
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        binding = DataBindingUtil.inflate(layoutInflater, R.layout.main_fragment_chat, container, false)
        registerChatCallback(chatCallback)
        initGui()
        return binding?.root
    }

    private fun initGui() {
        binding?.apply {
            chatSendButton.setOnClickListener {
                val user = chatContactEdittext.text.toString()
                val text = chatMessageEdittext.text.toString()
                if (user.isNotEmpty() && text.isNotEmpty()) {
                    if (!acManager.isRegisterState) {
                        Toast.makeText(context, R.string.no_registration, Toast.LENGTH_SHORT).show()
                    } else {
                        acManager.sendInstantMessage(user, text)
                    }
                }
            }
            if (AppUtils.getStringBoolean(requireContext(), R.string.enable_debug_mode)) {
                chatContactEdittext.setText(getString(R.string.sip_default_chat_contact))
                chatMessageEdittext.setText(getString(R.string.sip_default_chat_message))
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        unregisterChatCallback(chatCallback)
        binding = null
    }

    override fun onPauseFragment() {}
    override fun onResumeFragment() {}
}